src/Calculator/PerformanceOptimizedDelegateCalculator.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Money\Calculator;
+
+use Money\Calculator;
+
+/**
+ * Optimized `Money\Calculator` that delegates to either
+ * `BcMathCalculator` or `GmpCalculator` depending on the operation.
+ *
+ * `GmpCalculator` is approx 25% faster for add and subtract.
+ * `BcMathCalculator` is faster for all decimal operations.
+ *
+ * @psalm-immutable
+ */
+final class PerformanceOptimizedDelegateCalculator implements Calculator
```

Comment:
We probably need to expand https://github.com/moneyphp/money/tree/8953b873725c465bc55eee12a1c91b923e0d63c7/benchmark to include the raw calculators (as 3 almost-exactly-same benchmark classes)